packages/extension-core/src/domains/assetDiscovery/scanner.ts

```diff
@@ -192,42 +194,20 @@ class AssetDiscoveryScanner {
           for (const checks of chunkedChecks) {
             const res = await Promise.allSettled(
               checks.map(async (check) => {
-                const token = tokensMap[check.tokenId]
-                if (!token) {
-                  throw new Error(`No token found for tokenId ${check.tokenId}`)
-                }
-
-                if (token.type === "evm-erc20" || token.type === "evm-uniswapv2") {
-                  const balance = await client.readContract({
-                    abi: erc20Abi,
-                    address: token.contractAddress as EvmAddress,
-                    functionName: "balanceOf",
-                    args: [check.address as EvmAddress],
-                  })
-                  return balance.toString()
-                }
-
-                if (token.type === "evm-native") {
-                  const addressMulticall = client.chain?.contracts?.multicall3?.address
-                  if (addressMulticall) {
-                    // if multicall is available then fetch balance using this contract call,
-                    // this will allow the client to batch it along with other pending erc20 calls
-                    const balance = await client.readContract({
-                      abi: abiMulticall,
-                      address: addressMulticall,
-                      functionName: "getEthBalance",
-                      args: [check.address as EvmAddress],
-                    })
-                    return balance.toString()
+                try {
+                  const token = tokensMap[check.tokenId]
+                  if (!token) {
+                    throw new Error(`No token found for tokenId ${check.tokenId}`)
                   }
 
-                  const balance = await client.getBalance({
-                    address: check.address as EvmAddress,
-                  })
-                  return balance.toString()
+                  return await Promise.race([
+                    getEvmTokenBalance(client, token, check.address as EvmAddress),
+                    throwAfter(5000, "timeout"),
```

Comment:
Can we do
`throwAfter(5000, `Asset Scan Timeout - ${check.tokenId}`),`
instead? Since this will end up in Sentry